src/main/scala/com/github/sashjakk/spot/SpotRepo.scala

```diff
@@ -1,31 +1,34 @@
 package com.github.sashjakk.spot
 
-import cats.implicits.{catsSyntaxEitherId, catsSyntaxEq}
+import cats.effect.Sync
+import cats.implicits.catsSyntaxEq
 
 import java.util.UUID
 import scala.collection.mutable
 
-trait SpotRepo {
-  def create(spot: SpotCreate): Either[Throwable, Spot]
-  def findById(id: UUID): Option[Spot]
-  def findByIdentifier(identifier: String): Option[Spot]
+trait SpotRepo[F[_]] {
+  def create(spot: SpotCreate): F[Spot]
+  def findById(id: UUID): F[Option[Spot]]
+  def findByIdentifier(identifier: String): F[Option[Spot]]
 }
 
 object SpotRepo {
-  def inMemory(memory: mutable.Map[UUID, Spot] = mutable.Map.empty): SpotRepo =
-    new SpotRepo {
-      override def create(spot: SpotCreate): Either[Throwable, Spot] = {
-        val uuid = UUID.randomUUID()
-        val entity = Spot(uuid, spot.identifier, spot.userId)
+  def inMemory[F[_]: Sync](memory: mutable.Map[UUID, Spot] = mutable.Map.empty): SpotRepo[F] =
```

Comment:
If `F[_]: Sync` is ok here? Using just delay... maybe it should be more restrictive

Reply:
It's fine if you are going to replace it with a real base in a future, when you don't need to have sync here :)
Otherwise there are better alternatives to store state in memory.

Reply:
will be a database